lib/foreman/renderer.rb

```diff
@@ -107,7 +120,9 @@ def snippet(name, options = {})
         begin
           return unattended_render(template, nil, options[:variables] || {})
         rescue => exc
-          raise "The snippet '#{name}' threw an error: #{exc}"
+          e = ::Foreman::Exception.new(N_("The snippet '%{name}' threw an error: %{exc}"), { :name => name, :exc => exc })
```

Comment:
With the `rescue` in `unattended_render`, I think this rescue block is obsolete. If you remove it then it'll reduce the length and complexity of the error message:

    «ERF42-7327 [Foreman::Exception]: »The snippet 'kickstart_networking_setup' threw an error: ERF22-8069 [Foreman::Renderer::SyntaxError]: »Syntax error occurred while parsing the template, make sure you have all ERB tags properly closed and the Ruby syntax is valid. The Ruby error: kickstart_networking_setup:2: syntax error, unexpected [..]

Without this rescue block then it could simply be:

    ERF22-8069 [Foreman::Renderer::SyntaxError]: »Syntax error occurred while parsing the template, make sure you have all ERB tags properly closed and the Ruby syntax is valid. The Ruby error: kickstart_networking_setup:2: syntax error, unexpected [..]

which removes the unnecessary second error code and duplication in the message.